File: app/src/main/java/com/storehop/app/data/entity/StoreCategoryOrder.kt
package com.storehop.app.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index

@Entity(
    tableName = "store_category_order",
    primaryKeys = ["storeId", "categoryId"],
    foreignKeys = [
        ForeignKey(
            entity = Store::class,
            parentColumns = ["id"],
            childColumns = ["storeId"],
            onDelete = ForeignKey.CASCADE,
        ),
        ForeignKey(
            entity = Category::class,
            parentColumns = ["id"],
            childColumns = ["categoryId"],
            onDelete = ForeignKey.CASCADE,
        ),
    ],
    indices = [
        Index("categoryId"),
        Index("userId"),
        Index("deletedAt"),
    ],
)
data class StoreCategoryOrder(
    val storeId: String,
    val categoryId: String,
    val displayOrder: Int,
    val isSeeded: Boolean,
    val userId: String,
    val createdAt: Long,
    val updatedAt: Long,
    val deletedAt: Long?,
    @ColumnInfo(defaultValue = "1") val pendingSync: Boolean = true,
)

File: ios/Storehop/Data/Models/StoreCategoryOrder.swift
import Foundation
import GRDB

/// Per-store aisle ordering. One row per (store, category) the user has
/// drag-reordered. Categories without a row use a neutral fallback order
/// computed at query time (handled in ShoppingDao).
struct StoreCategoryOrder: Codable, FetchableRecord, MutablePersistableRecord, Hashable, Sendable {
    static let databaseTableName = "store_category_order"

    var storeId: String
    var categoryId: String
    var displayOrder: Int
    var isSeeded: Bool
    var userId: String
    var createdAt: Int64
    var updatedAt: Int64
    var deletedAt: Int64?
    var pendingSync: Bool

    enum Columns {
        static let storeId      = Column(CodingKeys.storeId)
        static let categoryId   = Column(CodingKeys.categoryId)
        static let displayOrder = Column(CodingKeys.displayOrder)
        static let isSeeded     = Column(CodingKeys.isSeeded)
        static let userId       = Column(CodingKeys.userId)
        static let createdAt    = Column(CodingKeys.createdAt)
        static let updatedAt    = Column(CodingKeys.updatedAt)
        static let deletedAt    = Column(CodingKeys.deletedAt)
        static let pendingSync  = Column(CodingKeys.pendingSync)
    }
}
